### PR TITLE
Add support for `db.query.summary` in SqlClientAttributesExtractor

### PR DESCRIPTION
**Pull request by trask**
Under stable semconv flag, adds support in SqlClientAttributesExtractor for `db.query.summary` and removes `db.operation.name` and `db.collection.name` where it was previously extracted from `db.query.text` due to stable semconv definitions:

> The operation name SHOULD NOT be extracted from `db.query.text`, when the database system supports query text with multiple operations in non-batch operations.

and

> The collection name SHOULD NOT be extracted from `db.query.text`, when the database system supports query text with multiple collections in non-batch operations.

This is all hidden in this PR behind a marker interface so we can roll it out afterwards per instrumentation in smaller PRs where we can evaluate and fine-tune its impact if needed.

Implementation notes:

- I made a copy of the jflex file to support stable semconv since there are significant savings to not tracking operation / collection name in the jflex file, and also to ensure we don't break old semconv with all of the new jflex code to support query summaries. Unfortunately, this means running both under `database/dup` setting, but I think it's acceptable that `dup` setting comes with some additional overhead (and we're still hopefully hitting cache most of the time).
- There are lots of (passing) edge case tests `ProblematicSqlQuerySummaryTest.java`. LLMs are really good at jflex.
- I've moved this PR to draft because most of it is vibe coded and because it's too big. Will work on splitting it up into multiple PRs.

### CODE REVIEW

File: instrumentation-api-incubator/src/main/jflex/SqlSanitizer.jflex [hunk 1]
```diff
@@ -365,62 +453,71 @@ WHITESPACE           = [ \t\r\n]+
...
             setOperation(new Create());
+            appendOperationToSummary("CREATE");
           }
           appendCurrentFragment();
           if (isOverLimit()) return YYEOF;
       }
   "DROP" {
           if (!insideComment) {
             setOperation(new Drop());
+            appendOperationToSummary("DROP");
```
Comment (laurit) [inline instrumentation-api-incubator/src/main/jflex/SqlSanitizer.jflex]: what if the query is `ALTER TABLE t2 DROP COLUMN c, DROP COLUMN d;` https://dev.mysql.com/doc/refman/8.4/en/alter-table.html In this case drop shouldn't count as an operation

File: instrumentation-api-incubator/src/main/jflex/SqlSanitizer.jflex [hunk 2]
```diff
@@ -365,62 +453,71 @@ WHITESPACE           = [ \t\r\n]+
...
             setOperation(new Select());
+            appendOperationToSummary("SELECT");
           }
           appendCurrentFragment();
           if (isOverLimit()) return YYEOF;
       }
   "INSERT" {
           if (!insideComment) {
             setOperation(new Insert());
+            appendOperationToSummary("INSERT");
```
Comment (laurit) [inline instrumentation-api-incubator/src/main/jflex/SqlSanitizer.jflex]: if I interpret https://www.postgresql.org/docs/current/sql-keywords-appendix.html correctly then postgers might allow using `INSERT` as a table or column name

**Comment by trask** [comment]
Closing in favor of #15971
